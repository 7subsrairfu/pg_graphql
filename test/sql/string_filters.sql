begin;
    create table memo(
        id int primary key,
        contents text
    );

    insert into memo(id, contents)
    values
        (1, 'foo'),
        (2, 'bar'),
        (3, 'baz');

    savepoint a;

    -- Filter by Int
    select jsonb_pretty(
        graphql.resolve($$
            {
              memoCollection(filter: {contents: {startsWith: "b"}}) {
                edges {
                  node {
                    id
                  }
                }
              }
            }
        $$)
    );
    rollback to savepoint a;

rollback;
