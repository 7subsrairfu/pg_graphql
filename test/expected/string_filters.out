begin;
    create table memo(
        id int primary key,
        contents text
    );
    insert into memo(id, contents)
    values
        (1, 'foo'),
        (2, 'bar'),
        (3, 'baz');
    savepoint a;
    -- Filter by Int
    select jsonb_pretty(
        graphql.resolve($$
            {
              memoCollection(filter: {contents: {startsWith: "b"}}) {
                edges {
                  node {
                    id
                  }
                }
              }
            }
        $$)
    );
          jsonb_pretty           
---------------------------------
 {                              +
     "data": {                  +
         "memoCollection": {    +
             "edges": [         +
                 {              +
                     "node": {  +
                         "id": 2+
                     }          +
                 },             +
                 {              +
                     "node": {  +
                         "id": 3+
                     }          +
                 }              +
             ]                  +
         }                      +
     }                          +
 }
(1 row)

    rollback to savepoint a;
rollback;
